### PR TITLE
feat: initial CI workflows

### PR DESCRIPTION
**Pull request by mjnagel**
Adds initial CI workflows following the Makefile pattern:
- deploys rke2 cluster
- deploys this package
- smoketest to validate PVCs can be provisioned (does not cover S3/RWX yet)
- release also runs tests before publishing
- release-please used for version management

Probably will switch to a bash script because make make me cry, just wanted to get this in first before blowing up the world.

Closes https://github.com/defenseunicorns/uds-package-rook-ceph/issues/4

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	zarf package create $(extra_create_args) --confirm
+
+.PHONY: deploy-zarf-package
+deploy-zarf-package: ## Deploy the zarf package.
+	zarf package deploy zarf-package-*.tar.zst --confirm
+
+.PHONY: publish-zarf-package
+publish-zarf-package: ## Publish the zarf package and skeleton.
+	zarf package publish zarf-package-*.tar.zst oci://ghcr.io/defenseunicorns/packages
+	zarf package publish . oci://ghcr.io/defenseunicorns/packages
```
Comment (andrewg-xyz) [inline Makefile:24]: I've not seen this before what is the intent?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	kubectl wait --for=jsonpath='{.status.phase}'=Bound -n test pvc/test-pvc
+	kubectl wait --for=condition=Ready -n test pod/test-pod --timeout=1m
+
+.PHONY: zarf-init
+zarf-init: ## Zarf init.
+	zarf tools download-init -a amd64
+	zarf init --confirm -a amd64
+
+.PHONY: create-cluster
+create-cluster: ## Create a test cluster with terraform
```
Comment (andrewg-xyz) [inline Makefile:39]: maybe rename to 'create-cloud-cluster' or something to indicate not for local

File: Makefile (hunk 1 above)
Comment (mjnagel) [reply Makefile:24]: Publishes the skeleton package in case someone wants to consume certain components/extend this package.

**Review by andrewg-xyz — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (mjnagel) [reply Makefile:39]: Good call, I can update this. I want to also get away from separate targets for CI/dev, and maybe figure out the local dev path, just need to get an amd64 system running.

Comment (mjnagel) [reply Makefile:39]: If you're okay with it I'll tackle this in a follow on PR, along with the other two things mentioned.

**Review by MxNxPx — APPROVED** [review]
:ship: :it:

**Pull request merged**